src/openforms/formio/dynamic_config/date.py

```diff
@@ -0,0 +1,143 @@
+    openForms: Optional[OpenFormsConfig]
+    datePicker: Optional[DatePickerConfig]
+
+
+DatetimeLike = Optional[Union[datetime, str]]
+
+
+@register("date")
+class DateComponent(BasePlugin):
+    """
+    Implement behaviour for our custom date component
+
+    In the JS, this component type inherits from Formio datetime component. See
+    ``src/openforms/js/components/form/date.js`` for the various configurable options.
+    """
+
+    verbose_name = _("Date component")
+
+    @classmethod
+    def mutate(cls, component: FormioDateComponent, data: DataMapping) -> None:
+        for key in ("minDate", "maxDate"):
+            config = cast(
+                DateConstraintConfiguration,
+                glom(component, f"openForms.{key}", default=None),
+            )
+            if config is None:
+                continue
+
+            if (mode := config.get("mode")) is None:
+                continue
+
+            # formio has set the datePicker.{key} value
+            if mode == "fixedValue":
+                continue
+
+            config = cls.normalize_config(config)
+            value = cls.calculate_delta(config, data)
+            if value and isinstance(value, datetime):
+                value = value.isoformat()
+            assign(component, f"datePicker.{key}", value, missing=dict)
+
+    @staticmethod
+    def normalize_config(
+        config: DateConstraintConfiguration,
+    ) -> DateConstraintConfiguration:
+        mode = config["mode"]
+        assert mode in ("future", "past", "relativeToVariable")
+        if mode == "relativeToVariable":
+            return config
+
+        # mode is now future or past -> convert that to a relative delta config
+        config["mode"] = "relativeToVariable"
+        config["variable"] = NOW_VARIABLE
+        include_today = cast(bool, glom(config, "includeToday", default=False))
+        config["operator"] = "add" if mode == "future" else "subtract"
+
+        delta = cast(
+            DateConstraintDelta,
+            {
+                "years": 0,
+                "months": 0,
+                "days": 0 if include_today else 1,
+            },
+        )
+        config["delta"] = delta
+        return config
+
+    @staticmethod
+    def calculate_delta(
+        config: DateConstraintConfiguration,
+        data: DataMapping,
+    ) -> DatetimeLike:
+        assert config["mode"] == "relativeToVariable"
+
+        base_value = cast(
+            Optional[datetime],
+            glom(data, config["variable"], default=None),
+        )
+        # can't do calculations on values that don't exist
+        if base_value is None:
+            return None
+
+        assert (
+            base_value.tzinfo is not None
+        ), "Expected the input variable to be timezone aware!"
+        base_date = timezone.localtime(value=base_value).date()
+
+        delta = relativedelta(
+            years=cast(int, glom(config, "delta.years", default=None) or 0),
```

Comment:
Why not `default=0` without the `or`?

Reply:
If the dict has `{"years": None}`, this results in `None` being passed to `relativedelta`, which then crashes. This normalizes missing and empty variables both alike to the int `0`.

Reply:
check.